dataset/shape.cpp

```diff
@@ -101,11 +101,11 @@ Dataset concatenate(const Dataset &a, const Dataset &b, const Dim dim) {
   // coords to dataset (which is not desirable for a variety of reasons). It is
   // unlikely that this will cause trouble in practice. Users can just use a
   // range slice of thickness 1.
-  auto result =
-      a.empty()
-          ? Dataset(std::map<std::string, Variable>(),
-                    concat(a.coords(), b.coords(), dim, a.sizes(), b.sizes()))
-          : Dataset();
+  Dataset result;
+  if (a.empty())
+    result.setCoords(
+        Coords(concatenate(a.sizes(), b.sizes(), dim),
+               concat(a.coords(), b.coords(), dim, a.sizes(), b.sizes())));
```

Comment:
Not the point of this PR, but why do we have `concatenate` and `concat`?

Reply:
`concat` is an implementation detail and not user-exposed.